pandas/core/generic.py

```diff
@@ -10349,6 +10349,39 @@ def _logical_func(
             )
             return res._logical_func(name, func, skipna=skipna, **kwargs)
 
+        if (
+            self.ndim > 1
+            and axis == 1
+            and len(self._mgr.arrays) > 1
+            and bool_only is not None
+        ):
+            # Try to avoid a potentially-expensive transpose
```

Comment:
can you throw here on down into a function and call it. (i now it has to be on self) but maybe _reduce_maybe_without_transpose or _reduce_fast or similar

Reply:
sure.  will need to do this to eventually extend this to sum and prod too.

Reply:
(xref #44793)